src/__tests__/calculate.test.js

```diff
@@ -0,0 +1,22 @@
+import calculate from '../components/logic/calculate';
+
+describe('Test the calculate operations', () => {
+  it('Test the AC button', () => {
+    const ACButton = calculate({}, 'AC');
+    expect(ACButton).toEqual({
+      total: null,
+      next: null,
+      operation: null,
+    });
+  });
+
+  it('Test 9 button properly when 3 and x are already recorded ', () => {
+    const ACBtn = calculate({ total: '3', next: null, operation: 'x' }, '9');
+    expect(ACBtn).toEqual({ total: '3', next: '9', operation: 'x' });
+  });
+
+  it('Test 8 button properly when 4.33 are already recorded', () => {
+    const btn = calculate({ next: '4.33', total: null }, '8');
+    expect(btn).toEqual({ next: '4.338', total: null });
+  });
+});
```

Comment:
- [ ] @JOdiemo and @H-Ramalan great work on adding tests for the calculate.js file, however you need to add all the test scenarios, check the if statements inside the method, each if statement is a scenario (the main if statement that its not inside another if statement).